src/EditorFeatures/Core/Implementation/TextBufferAssociatedViewService.cs

```diff
@@ -123,7 +123,9 @@ public static bool AnyAssociatedViewHasFocus(ITextBuffer textBuffer)
         }
 
         [Conditional("DEBUG")]
+#pragma warning disable CA1822 // Mark members as static - accesses instance members within #if DEBUG
```

Comment:
😕 Can you not just mark `OnTextViewClose` static at the same time?

Reply:
Ah, I did not get a CA1822 on that method and assumed it needs instance access. Seems the analyzer is bailing out on that method as it has event like signature. I can remove this pragma and make both methods static.